.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,144 @@
+          kubectl wait -n ${{ env.SUBSCRIPTION_NAMESPACE }} deployment/opendatahub-operator-controller-manager --for=condition=Available=true --timeout=60s
+        env:
+          CATALOG_SOURCE_NAME: "odh-olm-test"
+          CATALOG_SOURCE_NAMESPACE: "olm"
+          SUBSCRIPTION_NAME: "opendatahub-operator"
+          SUBSCRIPTION_NAMESPACE: "openshift-operators"
+
+      - name: Create Namespace for DSPO
+        run: |
+          kubectl create namespace ${{ env.DSPO_NS }}
+          kubectl config set-context --current --namespace=${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Prepare for Upgrade Testing
+        run: |
+          # Update the KfDef manifest with the latest released version
+          sed -i "s/main/${{ inputs.released-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+        working-directory: ${{ env.RESOURCES_DIR }}/kfdef
+
+      - name: Print KfDef Manifest Contents
+        run: cat ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+
+      - name: Deploy DSPO KfDef
+        run: |
+          kubectl apply -f ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml -n ${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: "data-science-pipelines-operator"
+
+      - name: Print ODH Operator Pod Logs
+        run: kubectl get pods -n openshift-operators -o jsonpath='{.items[*].metadata.name}' | xargs -I {} kubectl logs -n openshift-operators {}
+
+      - name: Wait for DSPO deployment to be ready
+        run: |
+          kubectl get deployments -n ${{ env.DSPO_NS }}
+          timeout 300 bash -c 'until [[ $(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}') ]]; do sleep 5 && echo "$(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}')"; done'
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Run pre-upgrade tests
+        run: |
+          chmod +x tests/upgrades/main.sh
+          ./tests/upgrades/main.sh
+        env:
+          DSPA_NS: test-ds-project-1
+          WORKING_DIR: ${{ github.workspace }}
+
+      - name: Prepare for Upgrade Testing
+        run: |
+          # Update the KfDef manifest with the candidate version
+          sed -i "s/main/${{ inputs.candidate-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+        working-directory: ${{ env.RESOURCES_DIR }}/kfdef
+
+      - name: Print KfDef Manifest Contents
+        run: cat ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+
+      - name: Deploy KfDef Core for the candidate DSP Version
+        run: |
+          kubectl apply -f ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml -n ${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Run upgrade tests
+        run: |
+          chmod +x tests/upgrades/main.sh
```

Comment:
same as before

Reply:
Done, updated the commit with these file permissions and removed it from here.